var/spack/repos/builtin/packages/opendatadetector/package.py

```diff
@@ -36,5 +36,4 @@ def cmake_args(self):
 
     def setup_run_environment(self, env):
         env.set("OPENDATADETECTOR_DATA", join_path(self.prefix.share, "OpenDataDetector"))
-        for d in self.libs.directories:
-            env.prepend_path("LD_LIBRARY_PATH", d)
+        # Note: ROOT dependency automatically sets up ROOT environment vars
```

Comment:
I think this won't work with DD4hep because it looks in `LD_LIBRARY_PATH` specifically and now this path would be in `ROOT_LIBRARY_PATH` instead

Reply:
Yes, DD4hep uses a similar mechanism as Gaudi to load plugins (i.e. it looks for  `.components` files on `LD_LIBRARY_PATH` and loads classes listed in there) which is separate from the ROOT plugin mechanism.

Reply:
@tmadlener @jmcarcell Thanks for the heads up on this. Looking at the `PluginServiceV{1,2}` under the Gaudi services, I see:
```c++
#if defined( _WIN32 )
          const std::string envVar = "PATH";
          const std::string sep    = ";";
#elif defined( __APPLE__ )
          const std::string envVar = "DYLD_LIBRARY_PATH";
          const std::string sep    = ":";
#else
          const std::string envVar = "LD_LIBRARY_PATH";
          const std::string sep    = ":";
#endif
```

Does this mean that the variable shouldn't simply be `LD` but be system dependent? Maybe it would be a good idea to have a similar path search to ROOT? I see `DD4HEP_LIBRARY_PATH` in a few places?

Reply:
> Does this mean that the variable shouldn't simply be LD but be system dependent?

Yes, I think so. (I haven't been involved in the implementations in either of these, so the usual grain of salt applies).

> Maybe it would be a good idea to have a similar path search to ROOT?

Yes, ideally it would be decoupled from the `LD_LIBRARY_PATH` (as also @wdconinc commented below). That will not happen on a timescale relevant for this PR. We probably need to bring this up with the developers to see whether they are open to something like this.

>  I see DD4HEP_LIBRARY_PATH in a few places?

From a quick grep through DD4hep, it looks like `DD4HEP_LIBRARY_PATH` is only picked up by (some of) the python wrappers and they then use it to set `DYLD_LIBRARY_PATH` via `os.environ`. I would interpret that to rather be a workaround than an actual solution.